scripts/us_census/api_utils/census_api_data_downloader.py

```diff
@@ -0,0 +1,472 @@
+                         len(csv_files_list[year]), year)
+            for csv_file in csv_files_list[year]:
+                cur_csv_path = os.path.join(output_path, csv_file)
+                print("Deleting", cur_csv_path)
+                logging.info('deleting %s', cur_csv_path)
+                if os.path.isfile(cur_csv_path):
+                    os.remove(cur_csv_path)
+
+
+def download_table_variables(dataset, table_id, year_list, geo_url_map_path,
+                             spec_path, output_path, api_key):
+    # TODO implement the method
+    pass
+
+
+#     table_id = table_id.upper()
+#     spec_dict = json.load(open(spec_path, 'r'))
+#     geo_url_map = json.load(open(geo_url_map_path, 'r'))
+
+#     output_path = os.path.expanduser(output_path)
+#     output_path = os.path.join(output_path, dataset)
+#     output_path = os.path.join(output_path, table_id+'_vars')
+#     os.makedirs(output_path, exist_ok=True)
+
+#     status_path = os.path.join(output_path, 'download_status.json')
+
+#     variables_year_dict = {}
+#     variable_col_map = get_yearwise_variable_column_map(dataset, table_id, year_list)
+#     print(list(variable_col_map))
+#     for year in year_list:
+#         variables_year_dict[year] = []
+#         for variable_id in variable_col_map[year]:
+#             column_name = variable_col_map[year][variable_id]
+#             t_flag = True
+#             if not column_to_be_ignored(column_name, spec_dict):
+#                 variables_year_dict[year].append(variable_id)
+#                 variables_year_dict[year].append(variable_id+'A')
+#         print(year)
+#         print(len(variables_year_dict[year]))
+
+#     url_list = get_variables_url_list(table_id, variables_year_dict, geo_url_map, output_path, api_key)
+#     url_list = sync_status_list([], url_list)
+#     with open(status_path, 'w') as fp:
+#         json.dump(url_list, fp, indent=2)
+
+#     print(len(url_list))
+#     logging.info("Compiled a list of %d URLs", len(url_list))
+
+#     start = time.time()
+
+#     rate_params = {}
+#     rate_params['max_parallel_req'] = 50
+#     rate_params['limit_per_host'] = 20
+#     rate_params['req_per_unit_time'] = 10
+#     rate_params['unit_time'] = 1
+
+#     failed_urls_ctr = download_url_list_iterations(url_list, url_add_api_key, api_key, async_save_resp_csv, url_filter=url_filter, rate_params=rate_params)
+
+#     with open(status_path, 'w') as fp:
+#         json.dump(url_list, fp, indent=2)
+
+#     # check status before consolidate, warn if any URL status contains fail
+#     if failed_urls_ctr > 0:
+#         logging.warn('%d urls have failed, output files might be missing data.', failed_urls_ctr)
+#     consolidate_files(dataset, table_id, year_list, output_path)
+
+#     end = time.time()
+#     print("The time required to download the", table_id, "dataset :", end-start)
+#     logging.info('The time required to download the %s dataset : %f', table_id, end-start)
```

Comment:
Can this block be removed?

Reply:
Have kept the block as it acts as a basic code which needs minor modification. The feature is not useful currently but would be helpful in future and would need minor modifications only. It was hard to implement and test it within the current timeframe.